desc/equilibrium/equilibrium.py

```diff
@@ -1866,21 +1878,22 @@ def _optimize(  # noqa: C901
         timer = Timer()
         timer.start("Total time")
 
-        eq = self
         if not objective.built:
-            objective.build(eq)
+            objective.build()
         if not constraint.built:
-            constraint.build(eq)
+            constraint.build()
 
-        cost = objective.compute_scalar(objective.x(eq))
+        cost = objective.compute_scalar(objective.x(self))
         perturb_options = deepcopy(perturb_options)
         tr_ratio = perturb_options.get(
             "tr_ratio",
             inspect.signature(optimal_perturb).parameters["tr_ratio"].default,
         )
 
         if verbose > 0:
-            objective.print_value(objective.x(eq))
+            objective.print_value(objective.x(self))
+
+        params = orig_params = self.params_dict.copy()
```

Comment:
this is just multiple assignment, it's equivalent to 
```python
params = self.params_dict.copy()
orig_params = self.params_dict.copy()
```

`params` will get modified during the optimization but we also keep track of the original params in case the optimization fails, or if the user doesn't want the equilibrium to be modified in place.